common/scripted_effects/babylon_country_implosion_effects.txt

```diff
@@ -0,0 +1,500 @@
+﻿babylon_country_implosion = {
+	# Scope = new country
+	add_journal_entry = {
+		type = je_babylon_political_cohesion
+	}
+	set_variable = is_cohesion_implosion_country
+	every_interest_group = {
+		limit = {
+			is_in_government = yes
+		}
+		remove_ruling_interest_group = yes
+	}
+	random_scope_pop = {
+		limit = {
+			root.capital.state_region = {
+				is_homeland = prev.culture
+			}
+		}
+		religion = {
+			save_scope_as = new_religion_scope
+		}
+		culture = {
+			save_scope_as = new_culture_scope
+		}
+	}
+	every_primary_culture = {
+		limit = {
+			NOT = {
+				this = scope:new_culture_scope
+			}
+		}
+		root = {
+			add_to_variable_list = {
+				name = deleted_cultures
+				target = prev
+			}
+		}
+	}
+	add_primary_culture = scope:new_culture_scope
+	every_in_list = {
+		variable = deleted_cultures
+		root = {
+			remove_primary_culture = prev
+		}
+	}
+	remove_list_variable = {
+		name = deleted_cultures
+	}
+	# remove list var
+	set_state_religion = scope:new_religion_scope
+	ordered_interest_group = {
+		order_by = interest_group_population
+		position = 1
+		max = 1
+		check_range_bounds = no
+		save_scope_as = dominant_ig_scope
+		add_ruling_interest_group = yes
+	}
+	every_interest_group = {
+		leader = {
+			kill_character = yes
+		}
+	}
+	scope:dominant_ig_scope = {
+		leader = {
+			set_character_as_ruler = yes
+		}
+	}
+	scope:dominant_ig_scope = {
+		switch = {
+			trigger = is_interest_group_type
+			ig_armed_forces = {
+				root = {
+					activate_law = law_type:law_presidential_republic
+					activate_law = law_type:law_professional_army
+					activate_law = law_type:law_outlawed_dissent
+					random_list = {
```

Comment:
So there is a event link introduced in 1.3.4 which is currently_active_law_in_group, can't you write something that chooses the most preferred law in law_group?

or use something like every_preferred_law and/or currently_active_law_in_group  and/or the prefers_law trigger

Reply:
Couldn't figure it out, I tried though.. In the end I bruteforced it.

Reply:
Because this will create nations with inherent political unrest because some laws will not match some flavor or non-generic ideologies. Very annoying to maintain too.